internal/controller/grafanaorganization_controller.go

```diff
@@ -384,34 +385,34 @@ func (r *GrafanaOrganizationReconciler) configureGrafana(ctx context.Context) er
 		},
 	}
 
-	_, err = controllerutil.CreateOrPatch(ctx, r.Client, grafanaConfig, func() error {
-		// We always sort the organizations to ensure the order is deterministic and the configmap is stable
-		// in order to prevent grafana to restarts.
-		slices.SortStableFunc(organizationList.Items, compareOrganizationsByID)
-
-		config, err := templating.GenerateGrafanaConfiguration(organizationList.Items)
-		if err != nil {
-			logger.Error(err, "failed to generate grafana user configmap values.")
-			return errors.WithStack(err)
-		}
-
-		// TODO: to be removed for next release
-		// cleanup owner references from the config map, see https://github.com/giantswarm/observability-operator/pull/183
-		for _, organization := range organizationList.Items {
-			// nolint:errcheck,gosec // ignore errors, owner references are probably already gone
-			controllerutil.RemoveOwnerReference(&organization, grafanaConfig, r.Scheme)
-		}
-
-		logger.Info("updating grafana-user-values", "config", config)
-
-		grafanaConfig.Data = make(map[string]string)
-		grafanaConfig.Data["values"] = config
+	// TODO remove after next release start
+	if err = r.Client.Delete(ctx, grafanaConfig); client.IgnoreNotFound(err) != nil {
+		return errors.WithStack(err)
+	}
 
-		return nil
-	})
+	// Retrieve the app.
+	var currentApp appv1.App = appv1.App{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "grafana",
+			Namespace: "giantswarm",
+		},
+	}
+	err = r.Client.Get(ctx, types.NamespacedName{Name: currentApp.GetName(), Namespace: currentApp.GetNamespace()}, &currentApp)
+	if err != nil {
+		return err
+	}
+	currentApp.Spec.UserConfig = appv1.AppSpecUserConfig{}
+	if err = r.Client.Update(ctx, &currentApp); err != nil {
+		return err
+	}
```

Comment:
I guess the part to be removed ends here?
Such a comment could help ensure we remove all we need.
```suggestion
	}
	// TODO end of section to be removed after next release (current: 0.10.1)
```

Reply:
Oh I added one initially so it was most likely lost :(